sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/implementation/util/BuilderHelper.java

```diff
@@ -195,12 +217,36 @@ private static HttpPipelinePolicy getResponseValidationPolicy() {
             .build();
     }
 
+    /**
+     * Validates that the client is properly configured to use https.
+     *
+     * @param objectToCheck The object to check for.
+     * @param objectName The name of the object.
+     * @param endpoint The endpoint for the client.
+     * @param logger {@link ClientLogger} used to log any exception.
+     */
+    public static void httpsValidation(Object objectToCheck, String objectName, String endpoint, ClientLogger logger) {
+        if (objectToCheck != null && !parseEndpoint(endpoint, logger).getScheme().equals(Constants.HTTPS)) {
+            throw new IllegalArgumentException("Using a(n) " + objectName + " requires https");
```

Comment:
Use logger.logAndThrow. Same to other places.

Reply:
Updated to use logger